st2common/st2common/services/action.py

```diff
@@ -548,3 +551,17 @@ def _is_notify_empty(notify_db):
     if not notify_db:
         return True
     return not (notify_db.on_complete or notify_db.on_success or notify_db.on_failure)
+
+
+def _is_notify_skipped(liveaction):
+    """
+    notification is skipped if action execution is under workflow context and
+    task is not specified under wf_ex_db.notify["tasks"].
+    """
+    if not workflow_service.is_action_execution_under_workflow_context(liveaction):
+        return False
+    wf_ex_db = WorkflowExecution.get_by_id(liveaction.workflow_execution)
```

Comment:
Just thinking out loud.

This means that we will now perform 2 additional DB queries for any workflow action when calling ``create_request``, right?

I guess that is probably not that big deal, but it may make sense to utilize "only_fields" so we only retrieve fields we need on those two models (``notify`` for ``WorkflowExecutionDB``, ``task_name`` for ``TaskExecution``).

---

EDIT: Yeah, ``WorkflowExecutionDB`` can also contain ``result`` and other large fields values so it is actually important we only retrieve fields we need.

Luckily my recent DB performance optimizations make this less of an issue (in the past, retrieving large fields such as results from the database could easily take many seconds or more), but we should still be careful and avoid retrieving unnecessary fields / data we don't need.

Reply:
Addressed.